### ISSUE TITLE
Upgrade to latest Flowershow (V2 - with MarkdownDB)

### ISSUE THREAD

**Issue by rufuspollock** [life-itself/community#402]
## Acceptance

- [ ] Upgrade to latest Flowershow

### Important changes

- E2E workflow added: since we decoupled our Flowershow template E2E tests from the template itself, these can now be used to test any Flowershow-based repository 
- [[...slug]] coming from the template needs to be replaced by [...slug] in each upgrade, as there is an index.tsx page as well

### Complications

- lots of customizations so almost no components from @portaljs/core would be used (if any) - unless we spend some time on refining the core package components and their API (advised) - otherwise, the package is hardly useful at all
- default frontmatter fields values defined in contentlayer.config.js - not supported in mddb yet, so would have to further customize our `computeFileds` lib we have in the template which is already a workaround/temporary solution...
  - this is mostly related to default layouts set in contentlayer.config.js for different doc types

### TODOs

- [ ] can we remove giscus.json from the template? why is it still there?
- [ ] is types.d.ts with @portaljs/remark-wiki-link declaration still needed? do we have an open issue for it in the portaljs repo?
- [ ] clean up .prettierignore file out of old patterns
- [ ] better way to generate site-wide toc - atm it's generated in each page
- [ ] review computeFields function - temporary solution we added for contentlayer's computed fields
```
        const dbFile = await mddb.getFileByUrl(path)
        const filePath = dbFile.file_path;
        const frontMatter = dbFile.metadata ?? {};

        const source = fs.readFileSync(filePath, { encoding: "utf-8" });
        const frontMatterWithComputedFields = await computeFields({
            frontMatter,
            urlPath: path,
            filePath,
            source,
        });
```
- [ ] better types for mddb - atm `await clientPromise` is of `any` type, so available methods are not visible
- [ ] remove Layout component from @portaljs/core - add it directly to the template (at /components/Layout.tsx)
- [ ] replace any hard coded colors from the core packes and replace with primary/secondary etc

**Comment by khalilcodes** [comment]
Flowershow latest has some layouts like navbar and footer in it's core package. Due to this it's not possible to have any sort of customization to those components. For example,

1. The current lifeitself site has custom elements in it's navbar like a 'join the conversation' button.
2. In the footer we have a signup newsletter form that loads on every page.

IMO it would be best to move these layouts out from core into template, which makes more sense for custom advanced websites.

**Comment by rufuspollock** [comment]
@khalilcodes i don't think we want to move anything much into core for now - we'll just to have to keep the overrides which is fine.

My sense here is that it is probably best if @olayway does a first pass as she has quite a bit of experience doing the upgrades. @olayway i think we look at this once you've done the upgrade to markdowndb in flowershow core etc.

**Comment by olayway** [comment]
@rufuspollock I tried upgrading using degit and going from the diffs as usual, but the Flowershow template used here is pretty old and it has been heavily customized. There are also some unsolved problems like handling original contentlayer's computed fields. I ended up with some working version of the project (can run-dev it) but there is still a long way to go.
What I could try instead of diffing is to only replace contentlayer with mddb if for some reason this is a must. But this is not a real upgrade, i.e. all the other elements that have been changed would not be updated. What I would advise instead is first:
1. improve mddb to support document types and computed fields (configurations similar to contentlayer ones)
2. improve our @portaljs/core package so that the components are more customizable and they don't have to be created anew for each Flowershow website anytime some customization is needed
3. review our template code and rethink its solutions in terms of ease of upgrades/customizations

Only then I would attempt to do the upgrade.
